source/application/wintaser.cpp

```diff
@@ -2885,177 +2874,79 @@ LPCWSTR ExceptionCodeToDescription(DWORD code)
 	}
 }
 
-static int CountSharedPrefixLength(const std::wstring& a, const std::wstring& b)
-{
-	int i = 0;
-	while(a[i] && tolower(a[i]) == tolower(b[i]))
-		i++;
-	return i;
-}
-static int CountBackslashes(const std::wstring& str)
-{
-    return std::count(str.begin(), str.end(), L'\\');
-}
-
-static int IsPathTrusted(const std::wstring& path)
-{
-	// we want to return 0 for all system or installed dlls,
-	// 2 for our own dll, 1 for the game exe,
-	// and 1 for any dlls or exes around the game's directory
-
-    if (!wcsicmp(path.c_str(), injected_dll_path.c_str()))
-    {
-        return 2;
-    }
-
-    int outCount = 2;
-
-	if(CountBackslashes(exe_filename.c_str() + CountSharedPrefixLength(exe_filename, path)) < outCount)
-		return 1;
-	if(CountBackslashes(sub_exe_filename.c_str() + CountSharedPrefixLength(sub_exe_filename, path)) < outCount)
-		return 1;
-
-    if(path.length() >= 4 && !wcsnicmp(path.c_str() + path.length() - 4, L".cox", 4)) // hack, we can generally assume the game outputted any dll that has this extension
-		return 1;
-
-	return 0;
-}
 
 
-struct MyMODULEINFO
-{
-	MODULEINFO mi;
-	std::wstring path;
-};
-static std::vector<MyMODULEINFO> trustedModuleInfos;
-static MyMODULEINFO injectedDllModuleInfo;
 
-void SetTrustedRangeInfo(TrustedRangeInfo& to, const MyMODULEINFO& from)
-{
-	DWORD start = (DWORD)from.mi.lpBaseOfDll;
-	to.end = start + from.mi.SizeOfImage;
-	to.start = start;
-}
 
-void SendTrustedRangeInfos(HANDLE hProcess)
-{
-	TrustedRangeInfos infos = {};
 
-	SetTrustedRangeInfo(infos.infos[0], injectedDllModuleInfo);
-	infos.numInfos++;
 
-	int count = trustedModuleInfos.size();
-	for(int i = 0; i < count; i++)
-	{
-		if(i+1 < ARRAYSIZE(infos.infos))
-		{
-			SetTrustedRangeInfo(infos.infos[1+i], trustedModuleInfos[i]);
-			infos.numInfos++;
-		}
-	}
-	
-	SIZE_T bytesWritten = 0;
-	WriteProcessMemory(hProcess, remoteTrustedRangeInfos, &infos, sizeof(infos), &bytesWritten);
-}
+static std::map<DWORD, std::vector<std::wstring>> trusted_modules;
 
-bool IsInRange(DWORD address, const MODULEINFO& range)
+static UINT CountSharedPrefixLength(const std::wstring& a, const std::wstring& b)
 {
-	return (DWORD)((DWORD)address - (DWORD)range.lpBaseOfDll) < (DWORD)(range.SizeOfImage);
+    size_t shortest_path = std::min(a.size(), b.size());
+    UINT i;
+    for (i = 0; i < shortest_path && tolower(a[i]) == tolower(b[i]); ++i);
+	return i;
 }
-bool IsInNonCurrentYetTrustedAddressSpace(DWORD address)
+static UINT CountBackslashes(const std::wstring& str)
 {
-	int count = trustedModuleInfos.size();
-	for(int i = 0; i < count; i++)
-		if(IsInRange(address, trustedModuleInfos[i].mi))
-			return true;
-	return (address == 0);
+    return std::count(str.begin(), str.end(), L'\\');
 }
 
-DWORD CalculateModuleSize(LPVOID hModule, HANDLE hProcess)
+static void ModuleLoaded(DWORD process_id, const std::wstring& path)

```

Comment:
Perhaps `OnModuleLoaded`? The current name can be confused with a function that checks if the given module is loaded.